processing/src/main/java/org/apache/druid/segment/join/JoinConditionAnalysis.java

```diff
@@ -133,26 +142,23 @@ public String getOriginalExpression()
    */
   public boolean isAlwaysFalse()
   {
-    return nonEquiConditions.stream()
-                            .anyMatch(expr -> expr.isLiteral() && !expr.eval(ExprUtils.nilBindings()).asBoolean());
+    return anyFalseLiteralNonEquiConditions;
   }
 
   /**
    * Return whether this condition is a constant that is always true.
    */
   public boolean isAlwaysTrue()
   {
-    return equiConditions.isEmpty() &&
-           nonEquiConditions.stream()
-                            .allMatch(expr -> expr.isLiteral() && expr.eval(ExprUtils.nilBindings()).asBoolean());
+    return equiConditions.isEmpty() && allTrueLiteralNonEquiConditions;
```

Comment:
It seems like `allTrueLiteralNonEquiConditions` is only used here; how about caching `isAlwaysTrue` directly?

Reply:
Done